pubtools/_quay/iib_operations.py

```diff
@@ -0,0 +1,270 @@
+        target_settings=target_settings,
+    )
+
+    _, tag = build_details.index_image.split(":", 1)
+    dest_image = image_schema.format(
+        host=target_settings.get("quay_host", "quay.io").rstrip("/"),
+        namespace=target_settings["quay_namespace"],
+        repo=get_internal_container_repo_name(target_settings["quay_operator_repository"]),
+        tag=tag,
+    )
+    # TODO: does target_settings["quay_namespace"] point to rh-osbs?
+    # Index image used to fetch manifest list. This image will never be overwritten
+    intermediate_index_image = image_schema.format(
+        host=target_settings.get("quay_host", "quay.io").rstrip("/"),
+        namespace=target_settings["quay_namespace"],
+        repo="iib",
+        tag=tag,
+    )
+
+    # Sign image
+    sig_handler = OperatorSignatureHandler(hub, task_id, target_settings)
+    sig_handler.sign_task_index_image(signing_key, intermediate_index_image, tag)
+
+    # Push image to Quay
+    # NOTE: tagging doesn't use intermediate index image, because we want the most up-to-date
+    #       image to be copied to the destination
+    ContainerImagePusher.run_tag_images(
+        build_details.index_image, [dest_image], True, target_settings
+    )
+
+
+def task_iib_remove_operators(
+    operators, archs, index_image, signing_key, hub, task_id, target_settings
+):
+    """
+    Perform all the necessary actions for the 'PushRemoveIIBOperators' entrypoint.
+
+    Args:
+        operators ([str]):
+            Operators to remove from the index image.
+        arch ([str]):
+            Architectures to build the index image for.
+        index_image (str):
+            Index image to remove the operators from.
+        signing_key (str):
+            Signing key to be used.
+        hub (HubProxy):
+            Instance of XMLRPC pub-hub proxy.
+        task_id (str):
+            ID of the pub task.
+        target_settings (dict):
+            Dictionary containing settings necessary for performing the operation.
+    """
+    image_schema = "{host}/{namespace}/{repo}:{tag}"
+    verify_target_settings(target_settings)
+
+    # Build new index image in IIB
+    build_details = OperatorPusher.iib_remove_operators(
+        operators=operators,
+        archs=archs,
+        index_image=index_image,
+        target_settings=target_settings,
+    )
+
+    _, tag = build_details.index_image.split(":", 1)
+    dest_image = image_schema.format(
+        host=target_settings.get("quay_host", "quay.io").rstrip("/"),
+        namespace=target_settings["quay_namespace"],
+        repo=get_internal_container_repo_name(target_settings["quay_operator_repository"]),
+        tag=tag,
+    )
+
+    # Index image used to fetch manifest list. This image will never be overwritten
+    intermediate_index_image = image_schema.format(
+        host=target_settings.get("quay_host", "quay.io").rstrip("/"),
+        namespace=target_settings["quay_namespace"],
+        repo="iib",
```

Comment:
This is not true for staging iib (it's iib-stage there). Emily pointed out in my code review in pub and it applies also here.

Reply:
Actually I was wrong, it's really iib. Only org is different per envs, which is correct here